### PR TITLE
Fix edit button not working correctly in extended configuration manager

### PR DESCRIPTION
**Pull request by Erftralle**
Fix for #102 .

### CODE REVIEW

File: administrator/components/com_joomgallery/controllers/config.php [hunk 1]
```diff
@@ -63,6 +63,14 @@ public function __construct()
...
+    $id  = JRequest::getInt('id');
+    $cid = JRequest::getVar('cid', array(), 'post', 'array');
+
+    if(!$id && count($cid) && $cid[0])
+    {
+      JRequest::setVar('id', (int) $cid[0]);
+    }
+
     JRequest::setVar('view', 'config');
     JRequest::setVar('hidemainmenu', 1);
```
Comment (jdgf) [inline administrator/components/com_joomgallery/controllers/config.php]: I have inserted following code after line 63 in config.php:

` 
   */


     public function edit()
   {
+    $id  = JRequest::getInt('id');
+    $cid = JRequest::getVar('cid', array(), 'post', 'array');
+
+    if(!$id && count($cid) && $cid[0])
+    {
+      JRequest::setVar('id', (int) $cid[0]);
+    }
+
     JRequest::setVar('view', 'config');
     JRequest::setVar('hidemainmenu', 1);



  /**

`

When opening configuration of joomgallery I have got a white monitor.

**Comment by MrMusic** [comment]
Tested with Joomla! 3.5.1/php 5.4 and with Joomla 3.6.0/php 7
Fix works fine for me :+1: 
Thank you!

File: administrator/components/com_joomgallery/controllers/config.php (hunk 1 above)
Comment (jdgf) [inline administrator/components/com_joomgallery/controllers/config.php]: Tested with Joomla 3.6 and php 5.6.

Comment (jdgf) [inline administrator/components/com_joomgallery/controllers/config.php]: Tested with Joomla 3.6 and php 7 I have got the notice: "0 syntax error, unexpected 'if' (T_IF) ".

Comment (Erftralle) [inline administrator/components/com_joomgallery/controllers/config.php]: @jdgf:
The code you inserted is far away from beeing valid PHP code.

Just insert
```
    $id  = JRequest::getInt('id');
    $cid = JRequest::getVar('cid', array(), 'post', 'array');

    if(!$id && count($cid) && $cid[0])
    {
      JRequest::setVar('id', (int) $cid[0]);
    }
```
at line 66.

**Comment by Erftralle** [comment]
@MrMusic 
Thanks for testing!

**Comment by Chraneco** [comment]
Thanks!

**Pull request merged**